### DISCUSSION TITLE
cmake sdl2

### DISCUSSION THREAD

**Discussion by darklightchaos** [openblack/openblack#298] [Q&A] [answered]
This is my firs time using visual studio I keep getting the sdl2 missing issue. I am not sure how to fix this. Can any one point me to the right direction?

**Comment by handsomematt** [comment]
On Windows you should simply use vcpkg, this should be the default behavior of just running CMake GUI it will download all dependencies for you automatically.

**Reply by darklightchaos** [reply]
I got this issue going now. :/

CMake Error at CMakeLists.txt:98 (set_property):
  set_property could not find TARGET bx.  Perhaps it has not yet been
  created.


CMake Error at CMakeLists.txt:99 (set_property):
  set_property could not find TARGET bimg.  Perhaps it has not yet been
  created.


CMake Error at CMakeLists.txt:100 (set_property):
  set_property could not find TARGET astc-codec.  Perhaps it has not yet been
  created.


CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:702 (_add_library):
  _add_library cannot create ALIAS target "bgfx::bgfx" because target "bgfx"
  does not already exist.
Call Stack (most recent call first):
  CMakeLists.txt:101 (add_library)


CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:664 (_add_executable):
  _add_executable cannot create ALIAS target "bgfx::shaderc" because target
  "shaderc" does not already exist.
Call Stack (most recent call first):
  CMakeLists.txt:102 (add_executable)


-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/ExternalProject.cmake:2650 (message):
  error: could not find git for clone of imgui-flame-graph-populate
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/ExternalProject.cmake:3700 (_ep_add_download_command)
  CMakeLists.txt:22 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
See also "C:/Users/esse_/Desktop/_deps/imgui-flame-graph-subbuild/CMakeFiles/CMakeOutput.log".

CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1005 (message):
  CMake step for imgui-flame-graph failed: 1
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1146:EVAL:2 (__FetchContent_directPopulate)
  C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1146 (cmake_language)
  C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1189 (FetchContent_Populate)
  CMakeLists.txt:115 (FetchContent_MakeAvailable)

**Reply by raffclar** [reply]
Looks like bgfx hasn't been set up correctly.

Did you clone the submodules too? The bgfx cmake config is a submodule. If you forgot to clone the submodules run `git submodule update --recursive`

**Comment by bwrsandman** [comment, marked as answer]
Refer to the README, the steps have been updated to make it more clear
